packages/grid/_modules_/grid/hooks/features/virtualization/useGridNoVirtualization.ts

```diff
@@ -0,0 +1,81 @@
+import * as React from 'react';
+import { GridComponentProps } from '../../../GridComponentProps';
+import { GridApiRef } from '../../../models/api/gridApiRef';
+import { useNativeEventListener } from '../../root/useNativeEventListener';
+import { useGridScrollFn } from '../../utils/useGridScrollFn';
+import { visibleGridColumnsSelector } from '../columns/gridColumnsSelector';
+import { useGridSelector } from '../core';
+import { useGridState } from '../core/useGridState';
+import { gridPaginationSelector } from '../pagination/gridPaginationSelector';
+
+export const useGridNoVirtualization = (
+  apiRef: GridApiRef,
+  props: Pick<GridComponentProps, 'disableVirtualization' | 'pagination' | 'paginationMode'>,
+): void => {
+  const windowRef = apiRef.current.windowRef;
+  const columnsHeaderRef = apiRef.current.columnHeadersElementRef;
+  const renderingZoneRef = apiRef.current.renderingZoneRef;
+  const [gridState, setGridState, forceUpdate] = useGridState(apiRef);
+  const [scrollTo] = useGridScrollFn(renderingZoneRef!, columnsHeaderRef!);
+  const paginationState = useGridSelector(apiRef, gridPaginationSelector);
+  const visibleColumns = useGridSelector(apiRef, visibleGridColumnsSelector);
+
+  const syncState = React.useCallback(() => {
+    if (!gridState.containerSizes || !windowRef?.current) {
+      return;
+    }
+
+    let firstRowIdx = 0;
+    const { page, pageSize } = paginationState;
+    if (props.pagination && pageSize != null && props.paginationMode === 'client') {
```

Comment:
`pageSize` can never be null here
See https://github.com/mui-org/material-ui-x/pull/2433/files/0eb2f693687f04990c2b318706cb4923ac25c91c#r695592921